app/controllers/mentors.js

```diff
@@ -0,0 +1,176 @@
+  const organisation = organisationModel.findOne({ organisationId: req.params.organisationId })
+  const mentors = mentorModel.findMany({ organisationId: req.params.organisationId })
+
+  mentors.sort((a, b) => {
+    return a.firstName.localeCompare(b.firstName) || a.lastName.localeCompare(b.lastName)
+  })
+
+  delete req.session.data.mentor
+
+  res.render('../views/mentors/list', {
+    organisation,
+    mentors,
+    actions: {
+      new: `/organisations/${req.params.organisationId}/mentors/new`,
+      view: `/organisations/${req.params.organisationId}/mentors`,
+      back: '/'
+    }
+  })
+}
+
+/// ------------------------------------------------------------------------ ///
+/// SHOW MENTOR
+/// ------------------------------------------------------------------------ ///
+
+exports.mentor_details = (req, res) => {
+  const organisation = organisationModel.findOne({ organisationId: req.params.organisationId })
+  const mentor = mentorModel.findOne({ mentorId: req.params.mentorId })
+
+  res.render('../views/mentors/show', {
+    organisation,
+    mentor,
+    actions: {
+      change: `/organisations/${req.params.organisationId}/mentors/${req.params.mentorId}/edit?referrer=change`,
+      delete: `/organisations/${req.params.organisationId}/mentors/${req.params.mentorId}/delete`,
+      back: `/organisations/${req.params.organisationId}/mentors`,
+      cancel: `/organisations/${req.params.organisationId}/mentors`
+    }
+  })
+}
+
+/// ------------------------------------------------------------------------ ///
+/// NEW MENTOR
+/// ------------------------------------------------------------------------ ///
+
+exports.new_mentor_get = (req, res) => {
+  let back = `/organisations/${req.params.organisationId}/mentors`
+  if (req.query.referrer === 'check') {
+    back = `/organisations/${req.params.organisationId}/mentors/new/check`
+  }
+
+  res.render('../views/mentors/find', {
+    mentor: req.session.data.mentor,
+    actions: {
+      save: `/organisations/${req.params.organisationId}/mentors/new`,
+      back,
+      cancel: `/organisations/${req.params.organisationId}/mentors`
+    }
+  })
+}
+
+exports.new_mentor_post = (req, res) => {
+  const errors = []
+
+  if (!req.session.data.mentor.trn.length) {
+    const error = {}
+    error.fieldName = 'mentor'
+    error.href = '#mentor'
+    error.text = 'Enter a teacher reference number (TRN)'
+    errors.push(error)
+  } else if (!validationHelper.isValidTRN(req.session.data.mentor.trn)) {
+    const error = {}
+    error.fieldName = 'mentor'
+    error.href = '#mentor'
+    error.text = 'Enter a valid teacher reference number (TRN)'
+    errors.push(error)
+  }
+
+  const mentor = mentorModel.findOne({
+    organisationId: req.params.organisationId,
+    trn: req.session.data.mentor.trn
+  })
+
+  if (mentor) {
+    const error = {}
+    error.fieldName = 'mentor'
+    error.href = '#mentor'
+    error.text = 'Mentor has already been added'
```

Comment:
'Mentor has already been added' isn't as natural as 'The mentor has already been added' (yes it's adding to character count, but as per GDS guidance, errors should sound natural)